build-gluster-org/scripts/user-update.py

```diff
@@ -0,0 +1,99 @@
+import requests
+import re
+import os
+from git import Repo
+import subprocess
+
+#Collects the gluster-maintainers names
+def create_user_data(repo_path):
+    os.chdir(repo_path)
+    username = "rakshithakamath94"
+    #Token will pe passed as secret text from jenkins
+    token = os.environ['TOKEN_RK']
+    user_data = requests.get('https://api.github.com/orgs/gluster/teams/gluster-maintainers/members', headers={'Authorization': 'token {}'.format(token)}).json()
+    return user_data
+
+#Function to find out the names which are missing in the jenkins conf file
+def search_pat(repo_path, filename, string_to_search,user_data):
+    os.chdir(repo_path)
+    INDENTATION_VALUE = ""
+    NOT_FOUND = []
+    for i, line in enumerate(open(filename)):
+        if string_to_search in line:
+            INDENTATION_VALUE=re.match(r"\s*", line).group()
+    for username in user_data:
+        with open(filename) as f:
+            if re.search(username['login'], f.read()):
+                continue
+            else:
+                NOT_FOUND.append("%s  "%INDENTATION_VALUE+"- "+username['login'])
+    f.close()
+    return NOT_FOUND
+
+#Function which appends the names which are not present in the jenkins conf file
+def edit_file(repo_path,filename,pattern,NOT_FOUND):
+    os.chdir(repo_path)
+    os.getcwd()
+    os.system('ls')
+    with open(filename, "r") as input_file:
+        buffer = input_file.readlines()
+
+    with open(filename, "w") as output_file:
+        search_pattern = re.compile(r'\s*%s'%pattern) 
+        for line in buffer:
+            if search_pattern.search(line):  
+                for names in NOT_FOUND:
+                    line = line + str(names) + "\n"
+            output_file.write(line)
+
+    input_file.close()
+    output_file.close()
+
+#Checks the files which are modified
+def check_untracked_files(repo_path,pat,branch_name):
+    os.chdir(repo_path)
+    FILES = []
+    os.system('git status')
+    out = subprocess.Popen(['git', 'status'], stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+    stdout, stderr = out.communicate()
+    list_me = stdout.decode('UTF-8').split('\n')
+    for line in list_me:
+         if pat in line:
+             val = line.split(':')
+             FILES.append(val[1].strip())
+    print (FILES)
+    if len(FILES) == 0:
+        print ("No changes")
+    else:
+        commit_and_push(FILES,branch_name)
+
+#Commit the files and push it to the repo
+def commit_and_push(FILENAMES,branch_name):
+    str_me=' '.join(FILENAMES)
+    print (str_me)
+    os.system('git config --global user.name "Rakshitha Kamath"')
```

Comment:
This would change the configuration for anyone running the script, who would then push stuff under your name, this seems like a problem, I think it should be removed.

Reply:
Again I am using the Gluster-ant username and password in the new PR submitted.